### PR TITLE
[PM-1839] Bug fix kdf modal

### PR DESCRIPTION
**Pull request by Dhruwang**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Fixed : [Change KDF modal cannot be closed in the expected way #5091
](https://github.com/bitwarden/clients/issues/5091)

In '[Encryption key settings](https://vault.bitwarden.com/#/settings/security/security-keys)', when we click on "Change KDF" , it triggers a modal to open.

In that modal there is a button at bottom named "cancel" which when clicked on, does nothing. Instead it should close the modal.

## Code changes

made changes in "change-kdf-confirmation.component.html"

Added bitDialogClose derivative to cancel button , which means that clicking on the button will close the dialog.

The [bit-dialog-close] input is set to false, which means that false will be passed back to the parent component when the dialog is closed.

Overall, this button will close the dialog and pass back false as the result when it is clicked.

## Screenshots

Before

https://user-images.githubusercontent.com/67850763/232026953-016e4737-2c96-4daa-8732-afdcbdc124f9.mp4

After 

https://user-images.githubusercontent.com/67850763/232027248-bcf4d514-a37a-4b71-a912-d418ae62e235.mp4

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Comment by Dhruwang** [comment]
@justindbaur can you please review this pr ?

File: apps/web/src/app/settings/change-kdf/change-kdf-confirmation.component.html [hunk 1]
```diff
@@ -43,7 +43,13 @@
     <button bitButton buttonType="primary" type="submit" [loading]="loading" form="form">
       <span>{{ "changeKdf" | i18n }}</span>
     </button>
-    <button bitButton buttonType="secondary" type="button" data-dismiss="modal">
+    <button
+      bitButton
+      buttonType="secondary"
+      type="button"
+      bitDialogClose
+      [bit-dialog-close]="false"
```
Comment (Hinton) [inline apps/web/src/app/settings/change-kdf/change-kdf-confirmation.component.html:51]: This shouldn't be needed.

```suggestion
```

**Review by Hinton — CHANGES_REQUESTED** [review]
One minor adjustment, adding `bitDialogClose` should be enough.

**Comment by Dhruwang** [comment]
Thank you @Hinton !
Committed changes

File: apps/web/src/app/settings/change-kdf/change-kdf-confirmation.component.html [hunk 2]
```diff
@@ -43,7 +43,12 @@
     <button bitButton buttonType="primary" type="submit" [loading]="loading" form="form">
       <span>{{ "changeKdf" | i18n }}</span>
     </button>
-    <button bitButton buttonType="secondary" type="button" data-dismiss="modal">
+    <button
+      bitButton
+      buttonType="secondary"
+      type="button"
+      bitDialogClose
+    >
```
[the remark this answers is not in the document]
Comment (Dhruwang) [reply apps/web/src/app/settings/change-kdf/change-kdf-confirmation.component.html:51]: Ya sure

**Pull request closed without merging**